### PR TITLE
feat: allow skipping of registry login command in verifyConditions

### PR DESCRIPTION
**Pull request by sjoukedv**
I need to push to artifactory which does not support the `helm registry login` command, but does work with `helm push`. With this new parameter I can disable the check.

Signed-off-by: Sjouke de Vries <info@sdvservices.nl>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,8 @@ appVersion 1.16.0
 | `populateChangelog` | `boolean` | `false` | `false`  | Populate `artifacthub.io/changes` annotations with notes produced by `@semantic-release/release-notes-generator` compatible plugins.  |
 | `packageArgs`       | `string`  | `""`    | `false`  | Additional parameters for the helm package command, e.g. `--key mykey --keyring ~/.gnupg/secring.gpg`                                 |
 | `cmPushArgs`        | `string`  | `""`    | `false`  | Additional parameters for the helm cm-push command (only relevant if `isChartMuseum` is set to true) e.g. `--context-path /repo/path` |
+| `skipRegistryLogin`        | `string`  | `""`    | `false`  | Additional parameter to skip the `helm registry login` in the verifyConditions step |
```
Comment (nflaig) [inline README.md:67]: this should be a boolean, see other boolean values for examples

File: lib/verifyConditions.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ module.exports = async (pluginConfig, context) => {
     if (registryHost && !pluginConfig.isChartMuseum && env.REGISTRY_USERNAME && env.REGISTRY_PASSWORD) {
         const registryUrl = registryHost.replace("oci://", "").split('/')[0];
         try {
-            await verifyRegistryLogin(registryUrl, env.REGISTRY_USERNAME, env.REGISTRY_PASSWORD);
+            await verifyRegistryLogin(registryUrl, env.REGISTRY_USERNAME, env.REGISTRY_PASSWORD, pluginConfig.skipRegistryLogin);
```
Comment (nflaig) [inline lib/verifyConditions.js:19]: why is verifyRegistryLogin called at all if skipRegistryLogin is true?

**Review by nflaig — CHANGES_REQUESTED** [review]

**Comment by sjoukedv** [comment]
> why is verifyRegistryLogin called at all if skipRegistryLogin is true? 

My bad, I am doing this besides my regular work. I have fixed the issue accordingly

**Comment by nflaig** [comment]
I also don't use this anymore so I don't have a good setup to quickly test changes... and since there are 0 tests and no linting / other static type checks changing anything is really error prone. Need to be careful not to break things.

File: README.md [hunk 3]
```diff
@@ -62,6 +62,7 @@ appVersion 1.16.0
 | `crConfigPath`      | `string`  | `""`    | `false`  | Path to .ct.yaml chart-releaser configuration file.                                                                                   |
 | `isChartMuseum`     | `boolean` | `false` | `false`  | Enable ChartMuseum publishing.                                                                                                        |
 | `populateChangelog` | `boolean` | `false` | `false`  | Populate `artifacthub.io/changes` annotations with notes produced by `@semantic-release/release-notes-generator` compatible plugins.  |
+| `skipRegistryLogin` | `boolean` | `false` | `false`  | Skip the `helm registry login` in the verifyConditions step                                                                           |
```
Comment (nflaig) [inline README.md:65]:
```suggestion
| `skipRegistryLogin` | `boolean` | `false` | `false`  | Skip the `helm registry login` command in the verifyConditions step                                                                           |
```

File: README.md [hunk 4]
```diff
@@ -62,6 +62,7 @@ appVersion 1.16.0
 | `crConfigPath`      | `string`  | `""`    | `false`  | Path to .ct.yaml chart-releaser configuration file.                                                                                   |
 | `isChartMuseum`     | `boolean` | `false` | `false`  | Enable ChartMuseum publishing.                                                                                                        |
 | `populateChangelog` | `boolean` | `false` | `false`  | Populate `artifacthub.io/changes` annotations with notes produced by `@semantic-release/release-notes-generator` compatible plugins.  |
+| `skipRegistryLogin` | `boolean` | `false` | `false`  | Skip the `helm registry login` command in the verifyConditions step                                                                           |
```
Comment (nflaig) [inline README.md:65]:
```suggestion
| `skipRegistryLogin` | `boolean` | `false` | `false`  | Skip the `helm registry login` command in the verifyConditions step.                                                                           |
```

**Review by nflaig — APPROVED** [review]

**Pull request merged**